src/api/api.c

```diff
@@ -654,14 +657,17 @@ typedef struct __z_reply_handler_wrapper_t {
 } __z_reply_handler_wrapper_t;
 
 void __z_reply_handler(_z_reply_t *reply, __z_reply_handler_wrapper_t *wrapped_ctx) {
+#if Z_FEATURE_QUERIES == 1
     z_owned_reply_t oreply = {._value = reply};
 
     wrapped_ctx->user_call(&oreply, wrapped_ctx->ctx);
     z_reply_drop(&oreply);  // user_call is allowed to take ownership of the reply by setting oreply._value to NULL
+#endif
 }
 
 int8_t z_get(z_session_t zs, z_keyexpr_t keyexpr, const char *parameters, z_owned_closure_reply_t *callback,
```

Comment:
If queries are disabled, I would completely omit any related APIs.

In doing so, the user will have an explicit error at compilation. Otherwise,  it will become harder for users (and us) to debug non-compliant behavior.

Also, you can save some extra bytes in the final binary.